person.rb

```diff
@@ -1,14 +1,24 @@
-class Person
-  attr_accessor :name, :age
-  attr_reader :id
+require_relative 'nameable'
+
+class Person < Nameable
+  attr_reader :id, :name, :age
 
   def initialize(age, name = 'Unknown', parent_permission: true)
+    super()
     @id = generate_id
     @name = name
     self.age = age
     @parent_permission = parent_permission
   end
 
+  def correct_name
+    @name
+  end
+
+  def age=(age)
+    @age = age.to_i
+  end
```

Comment:
- [ ] There is no need for this method to define `@age`, we can directly use `age` anywhere in this file.